app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours/ResurveyOpeningHours.kt

```diff
@@ -0,0 +1,78 @@
+package de.westnordost.streetcomplete.quests.opening_hours
+
+import android.util.Log
+import de.westnordost.osmapi.map.data.BoundingBox
+import de.westnordost.osmapi.map.data.Element
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.OsmTaggings
+import de.westnordost.streetcomplete.data.osm.OsmElementQuestType
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandler
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao
+import de.westnordost.streetcomplete.data.osm.tql.getQuestPrintStatement
+import de.westnordost.streetcomplete.data.osm.tql.toGlobalOverpassBBox
+import de.westnordost.streetcomplete.quests.DateUtil
+
+class ResurveyOpeningHours(private val overpassServer: OverpassMapDataDao) : OsmElementQuestType<OpeningHoursAnswer> {
+    override val commitMessage = "resurvey opening hours"
+    override val icon = R.drawable.ic_quest_opening_hours
+    override fun getTitle(tags: Map<String, String>) = R.string.resurvey_opening_hours_title
+    override fun getTitleArgs(tags: Map<String, String>, featureName: Lazy<String?>): Array<String> {
+        return arrayOf(tags["name"]!!)
+    }
+
+    override fun download(bbox: BoundingBox, handler: MapDataWithGeometryHandler): Boolean {
+        return overpassServer.getAndHandleQuota(getOverpassQuery(bbox)) { element, geometry ->
+            if (element.tags != null) {
+                // require opening hours that are supported
+                if (OpeningHoursTagParser.parse(element.tags["opening_hours"]!!) != null) {
+                    handler.handle(element, geometry)
+                } else {
+                    Log.wtf("AAAA", "opening_hours=" + element.tags["opening_hours"] + " was rejected as not representable in SC")
+                }
+            }
+
+        }
+    }
+
+    private fun getOverpassQuery(bbox: BoundingBox): String {
+        val groupName = ".old_opening_hours"
+        val nodeGroupName = groupName + "_nodes"
+        val wayGroupName = groupName + "_ways"
+        val relationGroupName = groupName + "_relations"
+        val reviewIntervalInDays = 380
+        return bbox.toGlobalOverpassBBox() + "\n" +
+                getQueryPart("node", nodeGroupName, reviewIntervalInDays) +
+                getQueryPart("way", wayGroupName, reviewIntervalInDays) +
+                getQueryPart("relation", relationGroupName, reviewIntervalInDays) +
+                "($nodeGroupName; $wayGroupName; $relationGroupName;);\n" +
+                getQuestPrintStatement()
+    }
+
+    // Note that newer segment will ensure that any edit,
+    // including adding or updating review marker like check_date or survey:date tags
+    // will cause OSM elements to become ineligible for this quest for reviewIntervalInDays days.
+    // It allows supporting check_date and any other survey markers without parsing of any tags.
```

Comment:
I don't understand this comment. What segment?

Reply:
Reformatted to

```
    /** Note that 'newer:' query part will ensure that any edit,
     * including adding or updating review marker like check_date or survey:date tags
     * will cause OSM elements to become ineligible for this quest for reviewIntervalInDays days.
     * It allows supporting check_date and any other survey markers without parsing of any tags.
     * In addition, objects where tag was surveyed and confirmed to be correct by adding check_date:opening_hours
     * are with larger delay, as it means this info on this specific object is relatively stable
     */
```

is it now more clear? It is above

```
    private fun getOverpassQuery(bbox: BoundingBox): String {
        val reviewIntervalInDays = 380
        val date = "'${DateUtil.getOffsetDateString(-reviewIntervalInDays * 3)}T00:00:00Z'"
        val dateTrippledTime = "'${DateUtil.getOffsetDateString(-reviewIntervalInDays * 3)}T00:00:00Z'"
        return bbox.toGlobalOverpassBBox() + """
            nwr[name][opening_hours]['opening_hours:signed'!='no']
                (if:
                    !is_date(t['check_date:opening_hours']) ||
                    date(t['check_date:opening_hours']) < date($dateTrippledTime)
                ) -> .old_opening_hours_tag;

            nwr[name][opening_hours](newer: $date) -> .recently_edited_objects_with_opening_hours;

            (.old_opening_hours_tag; - .recently_edited_objects_with_opening_hours;);
            """.trimIndent() + getQuestPrintStatement()
}
```

Reply:
Jup